examples/jsm/geometries/SDFGeometry.js

```diff
@@ -0,0 +1,319 @@
+      var s = 1.0 / e_count;
+      for(var i=0; i<3; ++i) {
+        v[i] = scale[i] * (x[i] + s * v[i]) + shift[i];
+      }
+      
+      //Add vertex to buffer, store pointer to vertex index in buffer
+      buffer[m] = vertices.length;
+      vertices.push(v);
+      
+      //Now we need to add faces together, to do this we just loop over 3 basis components
+      for(var i=0; i<3; ++i) {
+        //The first three entries of the edge_mask count the crossings along the edge
+        if(!(edge_mask & (1<<i)) ) {
+          continue;
+        }
+        
+        // i = axes we are point along.  iu, iv = orthogonal axes
+        var iu = (i+1)%3
+          , iv = (i+2)%3;
+          
+        //If we are on a boundary, skip it
+        if(x[iu] === 0 || x[iv] === 0) {
+          continue;
+        }
+        
+        //Otherwise, look up adjacent edges in buffer
+        var du = R[iu]
+          , dv = R[iv];
+        
+        //Remember to flip orientation depending on the sign of the corner.
+        if(mask & 1) {
+          faces.push([buffer[m],    buffer[m-du],    buffer[m-dv]]);
+          faces.push([buffer[m-dv], buffer[m-du],    buffer[m-du-dv]]);
+        } else {
+          faces.push([buffer[m],    buffer[m-dv],    buffer[m-du]]);
+          faces.push([buffer[m-du], buffer[m-dv],    buffer[m-du-dv]]);
+        }
+      }
+    }
+  }
+  
+  //All done!  Return the result
+  return { positions: vertices, cells: faces };
+};
+
+import { 
+    BufferAttribute, 
+    BufferGeometry, 
+    FloatType, 
+    Mesh, 
+    OrthographicCamera, 
+    PlaneGeometry, 
+    Scene, 
+    ShaderMaterial, 
+    Vector2, 
+    WebGLRenderer, 
+    WebGLRenderTarget 
+} from "three"
+
+let sdfCompute = `
+    varying vec2 vUv;
+    uniform float tileNum;
+    uniform float bounds;
+
+    [#dist#]
+
+    void main()	{ gl_FragColor=vec4( ( dist( vec3( vUv, tileNum ) * 2.0 * bounds - vec3( bounds ) ) < 0.00001 ) ? 1.0 : 0.0 ); }
+`
+
+let defaultDist = `
+    float dist( vec3 p ){
+        float c = length(p) - 0.5;
+        float comp = min( 1e10, c );
+        return comp; 
+    }
+`
+
+class SDFtoMesh extends BufferGeometry{
+	constructor( res = 64, distFunc = defaultDist, bounds = 1 ){
+		super()
+
+        this.renderer = new WebGLRenderer( { alpha : true, preserveDrawingBuffer : true, antialias : false } );
+        console.log( this.renderer )
+		const gl = document.createElement('canvas').getContext('webgl');
+        let maxTexSize = gl.getParameter(gl.MAX_TEXTURE_SIZE)
+
+		let w,h
+		if( res == 8 ) [w,h] = [ 32, 16]
+		else if( res == 16 ) [w,h] = [ 64, 64 ]
+		else if( res == 32 ) [w,h] = [ 256, 128 ]
+		else if( res == 64 ) [w,h] = [ 512, 512 ]
+		else if( res == 128 ) [w,h] = [ 2048, 1024 ]
+		else if( res == 256 ) [w,h] = [ 4096, 4096 ]
+		else if( res == 512 ) {
+			if (confirm("Are you sure you can handle 512? It might crash slower devices...")) [w,h] = [ 16384, 8096 ]
```

Comment:
Confirm dialogs do not belong to geometry generators.

Reply:
Yeah... generators should not talk that much.